### ISSUE TITLE
[Bug]: mo crash when loading tpcc 10 warehouse data in Pessimistic+RC mode

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#9856]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):8b3bb776a13d0a13625c77005959b60c2ee612bc
- Hardware parameters:16C 64G
- OS type:
- Others:
```

### Actual Behavior

when loading tpcc 10 warehouse data in Pessimistic+RC mode, the mo crash by:

2023/06/05 17:01:41.901432 +0800 INFO dn-service.logtail-push-service service/session.go:334 table not subscribed, filter out {"uuid": "dd4dccb4-4d3c-41f8-b482-5251dc7a41bf", "server-id": "d9db8ba8-740e-445b-a2d8-dfcd410a5029", "stream-id": 1, "remote": "127.0.0.1:58942", "id": "db_id:272448 tb_id:272462 ", "checkpoint": ""}
2023/06/05 17:01:41.901658 +0800 ERROR frontend/mysql_cmd_executor.go:3481 error: internal error: panic BUG: invalid ref count, -0xc0186f4240(-1):
github.com/matrixorigin/matrixone/pkg/lockservice.(*waiter).unref
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/lockservice/waiter.go:131
github.com/matrixorigin/matrixone/pkg/lockservice.(*waiter).close
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/lockservice/waiter.go:311
github.com/matrixorigin/matrixone/pkg/lockservice.(*localLockTable).unlock.func1
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/lockservice/lock_table_local.go:157
github.com/matrixorigin/matrixone/pkg/lockservice.(*fixedSlice).iter
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/lockservice/slice.go:159
github.com/matrixorigin/matrixone/pkg/lockservice.(*localLockTable).unlock
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/lockservice/lock_table_local.go:153
github.com/matrixorigin/matrixone/pkg/lockservice.(*activeTxn).close
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/lockservice/txn.go:149
github.com/matrixorigin/matrixone/pkg/lockservice.(*service).Unlock
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/lockservice/service.go:109
github.com/matrixorigin/matrixone/pkg/txn/client.(*txnOperator).unlock
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/txn/client/operator.go:789
github.com/matrixorigin/matrixone/pkg/txn/client.(*txnOperator).doWrite
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/txn/client/operator.go:497
github.com/matrixorigin/matrixone/pkg/txn/client.(*txnOperator).Commit
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/txn/client/operator.go:368
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnHandler).CommitTxn
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/frontend/txn.go:288
github.com/matrixorigin/matrixone/pkg/frontend.(*Session).TxnCommitSingleStatement
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/frontend/txn.go:583
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).executeStmt.func1
        /mnt/datadisk0/actions-runner/_work/mo-nightly-regression/mo-nightly-regression/matrixone/pkg/frontend/mysql_cmd_executor.go:2435
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).executeStmt.func3

mo log:
[mo-service.log](https://github.com/matrixorigin/matrixone/files/11651331/mo-service.log)

### Expected Behavior

MO SHOULD NEVER CRASH

### Steps to Reproduce

```Markdown
just run a mo Pessimistic+RC mode, and load tpcc 10 warehouse data using load data command
```

**Comment by volgariver6** [comment]
please check this issue @zhangxu19830126

**Comment by zhangxu19830126** [comment]
not woking on this

**Comment by zhangxu19830126** [comment]
本地找到复现的办法，找到问题

**Comment by zhangxu19830126** [comment]
fixed

**Comment by aressu1985** [comment]
fixed:nightly-d4d74e7
